### ISSUE TITLE
TransactionFailureException should be TransientError

### ISSUE THREAD

**Issue by aseemk** [neo4j/neo4j#5257]
When we add a constraint in production (Neo4j 2.2.3), we frequently see `Neo.DatabaseError.Transaction.CouldNotCommit` errors with messages like this:

```
java.lang.RuntimeException: org.neo4j.kernel.api.exceptions.TransactionFailureException:
Database constraints have changed (txId=61266153) after this transaction (txId=61266152) started,
which is not yet supported. Please retry your transaction to ensure all constraints are executed.
```

That's understandable, but shouldn't these errors be `TransientError`s then, not `DatabaseError`? Our code already supports automatically retrying on `TransientError`s.

Thanks!

/cc @ryanwe

**Comment by ryanwe** [comment]
Agreed it should be a `TransientError`. The error message even recommends retrying. And for the time being, I think it's safe to add `Neo.DatabaseError.Transaction.CouldNotCommit` to our list of retry errors. Want to open an issue in Neo4j?

**Comment by tinwelint** [comment]
That's true, this should be fixed.

**Comment by lutovich** [comment]
Hello @aseemk,

This issue was fixed by commit https://github.com/neo4j/neo4j/commit/b33cc35 and this is a test that verifies the fix https://github.com/neo4j/neo4j/commit/b33cc35.
Next 2.2 and 2.3 releases should have it.

**Comment by tinwelint** [comment]
FYI this fix is already available in 2.2.4, but of course, you can might as well use 2.2.5 if you're changing version to begin with

**Comment by aseemk** [comment]
Awesome, thanks folks!
